packages/application-components/src/components/modal-pages/internals/modal-page.tsx

```diff
@@ -125,57 +117,69 @@ const ModalPage = ({
-          onRequestClose={handleClose}
-          shouldCloseOnOverlayClick={Boolean(props.onClose)}
-          shouldCloseOnEsc={Boolean(props.onClose)}
-          overlayElement={getOverlayElement}
-          overlayClassName={{
-            base: makeClassName(getOverlayStyles({ size, ...props })),
-            afterOpen: makeClassName(getAfterOpenOverlayAnimation()),
-            beforeClose: makeClassName(getBeforeCloseOverlayAnimation()),
-          }}
-          className={{
-            base: makeClassName(getContainerStyles({ size, ...props })),
-            afterOpen:
-              typeof props.afterOpenStyles === 'string'
-                ? props.afterOpenStyles
-                : makeClassName(
-                    props.afterOpenStyles ?? getAfterOpenContainerAnimation()
-                  ),
-            beforeClose: makeClassName(
-              getBeforeCloseContainerAnimation({ size, ...props })
-            ),
-          }}
-          contentLabel={props.title}
-          parentSelector={getParentSelector}
-          ariaHideApp={false}
-          // Adjust this value if the (beforeClose) animation duration is changed
-          closeTimeoutMS={TRANSITION_DURATION}
-          style={{
-            // stylelint-disable-next-line selector-type-no-unknown
-            overlay: {
-              zIndex: props.zIndex,
-            },
-          }}
+    <DialogRoot
+      open={props.isOpen && !forceClose}
+      onOpenChange={handleClose as unknown as DialogProps['onOpenChange']}
+      modal={false} // `true` would apply aria-hidden to all other elements when the modal is open
+    >
+      <DialogPortal container={portalContainer}>
+        <div
+          // Assign the `data-role` to the overlay container, which is used as
+          // the CSS selector in the `<PortalsContainer>`.
+          data-role="modal-overlay"
         >
-          {!props.hideTopBar && (
-            <ModalPageTopBar
-              color={props.topBarColor}
-              onClose={handleClose}
-              currentPathLabel={props.currentPathLabel}
-              previousPathLabel={props.previousPathLabel}
-              hidePathLabel={props.hidePathLabel}
-            />
-          )}
-          {props.children}
-        </Modal>
-      )}
-    </ClassNames>
+          <ModalOverlay
+            size={size}
+            data-role="modal-overlay-clickable"
+            onClick={handleClose}
+          />
+          <ModalContent
+            size={size}
+            aria-describedby={undefined}
+            onInteractOutside={(e) => {
+              // Prevent only the modal from closing, handle all other events
+              const overlay = document.querySelector(
+                '[data-role="modal-overlay-clickable"]'
+              );
+              if (!overlay?.contains(e.target as Node)) {
+                e.preventDefault();
+              }
+            }}
+            aria-labelledby={props.title}
```

Comment:
Nit: the `aria-labelledby` should reference another element. We should remove that, as we already have `aria-label`.